### DISCUSSION TITLE
Apostrophe 2.115.0, media library extension, and a Gatsby source plugin for A3

### DISCUSSION THREAD

**Discussion by localghost8000** [apostrophecms/apostrophe#2723] [Release Notes]
For this release, we've made some minor updates to the 2.x core and also have 2 new modules, our Gatsby source plugin for Apostrophe 3, and a new media library integration for Apostrophe 2.x. Check out [part one](https://apostrophecms.com/blog/building-a-gatsby-site-with-apostrophecms) and [part two](https://apostrophecms.com/blog/building-a-gatsby-site-with-apostrophecms-data-part-2) of our tutorial write up on the Gatsby source plugin, written by @abea, and stay tuned for another announcement regarding the media library integration, which allows you to pull in images from Unplash and Wedia and adds a number of filter options for images (portrait, landscape, etc)

**apostrophe (core) 2.115.0**

- When `enableAltField` option is set to `true`, add an `alt` field to `apostrophe-images` schema. And use it in `apostrophe-images-widget`, in `<img>` alt attributes.

**[gatsby-source-apostrophe 1.0.0-beta](https://github.com/apostrophecms/gatsby-source-apostrophe)**

- Initial beta release of a source plugin for accessing [ApostropheCMS 3.x content APIs](https://a3.docs.apos.dev/guide/rest-apis.html) in [Gatsby](https://www.gatsbyjs.com/docs/tutorial/)

**[apostrophe-media-sources 1.0.0](https://github.com/apostrophecms/apostrophe-media-sources)**

- Initial release of a plugin that is the foundation for connecting other media sources to the Apostrophe Media Library
- Includes Unsplash integration that allows editors to easily search Unsplash and import images into the Apostrophe Media Library

**Comment by boutell** [comment]
Excited to see apostrophe-media-sources! BTW, it also supports wedia
integration.

On Wed, Feb 10, 2021 at 3:45 PM Ben Loeffler <notifications@github.com>
wrote:

> For this release, we've made some minor updates to the 2.x core and also
> have 2 new modules, our Gatsby source plugin for Apostrophe 3, and a new
> media library integration for Apostrophe 2.x. Check out part one
> <https://apostrophecms.com/blog/building-a-gatsby-site-with-apostrophecms>
> and part two
> <https://apostrophecms.com/blog/building-a-gatsby-site-with-apostrophecms-data-part-2>
> of our tutorial write up on the Gatsby source plugin, written by @alex
> <https://github.com/alex> Bea, and stay tuned for another announcement
> regarding the media library integration, which allows you to pull in images
> from Unplash and Wedia and adds a number of filter options for images
> (portrait, landscape, etc)
>
> *apostrophe (core) 2.115.0*
>
>    - When enableAltField option is set to true, add an alt field to
>    apostrophe-images schema. And use it in apostrophe-images-widget, in
>    <img> alt attributes.
>
> *gatsby-source-apostrophe 1.0.0-beta
> <https://github.com/apostrophecms/gatsby-source-apostrophe>*
>
>    - Initial beta release of a source plugin for accessing ApostropheCMS
>    3.x content APIs <https://a3.docs.apos.dev/guide/rest-apis.html> in
>    Gatsby <https://www.gatsbyjs.com/docs/tutorial/>
>
> *apostrophe-media-sources 1.0.0
> <https://github.com/apostrophecms/apostrophe-media-sources>*
>
>    - Initial release of a plugin that is the foundation for connecting
>    other media sources to the Apostrophe Media Library
>    - Includes Unsplash integration that allows editors to easily search
>    Unsplash and import images into the Apostrophe Media Library
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apostrophecms/apostrophe/discussions/2723>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAH27JQ5OBCIPJ6XEWF7ODS6LV6NANCNFSM4XNURGYA>
> .
>


-- 

THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
APOSTROPHECMS | apostrophecms.com | he/him/his

**Reply by ashleigham** [reply]
Excited to put apostrophe-media-sources to work!

**Comment by nikolalakovic85** [comment]
Hi @localghost443  and @abea I started with Gatsby tutorial, and I have some questions. 

1. Should I add in app.js property "baseUrl": "http://localhost:3000" since terminal is throwing this error _(Error fetching rendered page HTML. If related to not having an absolute URL, check if `baseUrl` is set on the Apostrophe app. TypeError: Only absolute URLs are supported)_, when I add "baseUrl" property this error is gone. Maybe this information should be provided in the Gatsby tutorial.
2. Second thing is, when I go to graphQL query page I can't see anything from apos for program and member stuff (screenshot attached), though allAposCorePage is available. Any hint on this one? (I installed and configured 'gatsby-source-apostrophe' plugin according to your instructions.)

![Screen Shot 2021-02-11 at 8 37 44 PM](https://user-images.githubusercontent.com/57252231/107690220-1fe0d300-6caa-11eb-8962-83950f95cb40.png)

I tried GET request via Postman and 'results' are zero.
![Screen Shot 2021-02-11 at 9 12 50 PM](https://user-images.githubusercontent.com/57252231/107693286-3721bf80-6cae-11eb-97c2-71744e37ae22.png)

**Reply by abea** [reply]
Thank you for trying the plugin, @nikolalakovic85.

1. Yes, you are correct. I automatically add the `baseUrl` setting in my `data/local.js` file, but I didn't make that clear and that file isn't committed to the repo. I'll update the headless demo to fix this.
2. In Postman you can see that you get a successful response (a `200` response with data), but there are simply no staff members to show. If you haven't yet added logged into the Apostrophe site and added some demo content, that should be all you need to do (for both programs and staff members).

By the way, you may want to update the `package.json` file to use `"apostrophe": "3.0.0-alpha.3"` rather than the Github source. We made an update to core recently that updates the version of webpack, which will effect the UI build from the demo (we are in alpha, after all). That `3.0.0-alpha.3` version will work with the demo from the blog post.

**Reply by nikolalakovic85** [reply]
Awesome @abea! I will try it tommorow! Thank you!

**Reply by nikolalakovic85** [reply]
Great @abea! It works, demo is fully functional. Can I try to recreate "Museum" theme with Gatsby? What do you think is it possible?

**Reply by abea** [reply]
You're certainly welcome to. My reason for using Tailwind in the demo was to make it easy to share styles between the Apostrophe app and the Gatsby site. A few thoughts around that:

Styles do not at all need to be exactly the same between the two, but basic layout should probably be the same so editors have a decent idea what the end result will look like (e.g., two column widget should be two columns everywhere). So the museum styles don't need to be complete in both, but just enough so that editign works. (you're only playing around with this so far, so copying a CSS file between the two is also fine)

I'm not a total CSS framework fan, and frankly am not in love with Tailwind, but if I were doing this in production I would probably separate my layout styles from "theme" styles somehow, then I could work out a way to share the layout styles and keep the theme styles only in Gatsby.

Also, the Open Museum demo site theme is fairly complex, as is that site as a whole (we have given more thought on how to approach demos like that in the future). You probably know this already, but it'd likely be easier to write new styles _inspired by_ Open Museum rather than try to use the CSS from that demo.

Good luck! We'd love to hear your thoughts as you work with it, as well as the source plugin and A3 in general.

**Reply by nikolalakovic85** [reply]
Thanks for answering, I will get back with my results!

**Reply by nikolalakovic85** [reply]
Hello @abea  after updating to "apostrophe: 3.0.0-alpha.3" I was forced to manually install "lowlight" library and I got this error
  **45:11  error  Cannot query field "_rendered" on type "AposStaffMemberPhoto"  graphql/template-strings**
I was looking into grapql schema and _rendered property is gone now! I know that this is still alpha but I wanted to appoint to this error.

**Reply by abea** [reply]
Would you create an issue for this error? I have not seen the install requirement before, but it also sounds like we might need to add some error coverage.

Please create the issue on this repo: https://github.com/apostrophecms/gatsby-source-apostrophe/

**Reply by nikolalakovic85** [reply]
Sure, I will do it!

**Comment by nikolalakovic85** [comment]
Another question regarding important feature of Gatsby which is image transformation, I'm currently investigating how can I use this with ApostropheCMS, is there an option for this already @abea ?

**Reply by abea** [reply]
There is not an option for image transformation yet. It seemed beyond the scope for the initial beta version. I'll add it to the list of future features -- to explore it, at least.

**Reply by nikolalakovic85** [reply]
maybe this can help https://mcro.tech/blog/gatsby-image-sharp/
